### PR TITLE
Feat/27

### LINKED ISSUE #18: User Story 27
Admin Merchant Enable/Disable

As an admin,
When I visit the admin merchants index (/admin/merchants)
Then next to each merchant name I see a button to disable or enable that merchant.
When I click this button
Then I am redirected back to the admin merchants index
And I see that the merchant's status has changed

### PR DESCRIPTION
**Pull request by RodrigoACG**
In this branch user story 27 was implemented. During this user story I had made a button to enable or disable a merchant. Also I included a sad path for user story 26.

close #18

### CODE REVIEW

File: app/views/admin/merchants/index.html.erb [hunk 1]
```diff
@@ -1,5 +1,13 @@
 <h1>Merchants</h1>
 
 <% @merchants.each do |merchant| %>
-<%= link_to "#{merchant.name}", admin_merchant_path(merchant.id), method: :get %>
+  <div id="merchant-<%= merchant.id %>">
```
Comment (LJ9332) [inline app/views/admin/merchants/index.html.erb:4]: Glad to see you figured out this within block

File: db/migrate/20240412222947_add_status_to_merchants.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToMerchants < ActiveRecord::Migration[7.1]
+  def change
+    add_column :merchants, :status, :integer, default: 0
```
Comment (LJ9332) [inline db/migrate/20240412222947_add_status_to_merchants.rb:3]: I will dm in slack for every one to run a nuke (rails db:{drop, create, migrate, seed}) command in the terminal.

File: app/models/merchant.rb [hunk 3]
```diff
@@ -5,5 +5,7 @@ class Merchant < ApplicationRecord
   has_many :transactions, through: :invoices
   has_many :customers, through: :invoices
 
+  enum status: ["Enabled", "Disabled"]
```
Comment (LJ9332) [inline app/models/merchant.rb]: can you refactor this to be a hash (i.e. {"Enabled => 0, "Disabled" => 1}) so if we have to update the merchant status there is a easy place to find what number is tied to what status.

Comment (RodrigoACG) [reply app/models/merchant.rb]: Fixed that issue

**Pull request merged**